app/src/main/java/com/android/unio/model/event/EventListViewModel.kt

```diff
@@ -56,6 +56,20 @@ class EventListViewModel(private val repository: EventRepository) : ViewModel()
     }
   }
 
+  /**
+   * Finds an event in the event list by its ID.
+   *
+   * @param id The ID of the event to find.
+   * @return The event with the given ID, or null if no such event exists.
+   */
+  fun findEventById(id: String): Event? {
```

Comment:
This is a good function to have, but as mentionned in a previous comment, consider using the `EventViewModel` for your pull request. It contains a single `Event`, just like the `UserViewModel`, and should be instanciated when you navigate to an event's details page.